internal/dao/stubs/account_dao.go

```diff
@@ -88,7 +88,7 @@ func (stub *accountDaoStub) GetByOrgId(ctx context.Context, orgId string) (*mode
 
 func (stub *accountDaoStub) GetByAccountNumber(ctx context.Context, number string) (*models.Account, error) {
 	for _, acc := range stub.store {
-		if *acc.AccountNumber == number {
+		if acc.AccountNumber.Valid && acc.AccountNumber.String == number {
 			return acc, nil
 		}
 	}
```

Comment:
Can you add a test when account number is actually NULL and test that? For both insert and select statements.

Reply:
Tests are in this PR: https://github.com/RHEnVision/provisioning-backend/pull/127/files
I could rebase and make one PR out of these two, but I thought maybe it might be better idea to keep these two separated.

Reply:
I would rather see unit tests executing all edge cases (nulls, empty values etc) and integration tests to focus on SQL statements and data. Also, I would like to ensure this patch is correct as part of the PR.